src/cdn_definitions/_impl/__init__.py

```diff
@@ -1,26 +1,88 @@
 import json
 import os
+import warnings
+
+import requests
 import yaml
 
 
-def load_data():
-    # Load data from first existing of these
-    candidate_paths = [
-        os.path.join(os.path.dirname(os.path.dirname(__file__)), "data.yaml"),
-        "/usr/share/cdn-definitions/data.yaml",
-    ]
+def get_data(data_paths):
+    """Loads data from YAML or JSON format into a dictionary.
+
+    Args:
+        data_paths (list[str]): A list of local paths or URLs to a JSON or YAML data file.
+
+    Returns:
+        dict: The data from the local path or URL in the form of a dictionary.
+
+    Raises:
+        RuntimeError: If all attempted data paths are invalid, a RuntimeError will be raised.
+
+    """
+    for p in data_paths:
```

Comment:
You might consider splitting this into some smaller functions, there's a lot of conditionals here.